src/csharp/Microsoft.Spark/Sql/Types/Timestamp.cs

```diff
@@ -0,0 +1,143 @@
+                    .AddTicks(microsecond * 10).ToUniversalTime();
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException($"Invalid microsecond value {microsecond}");
+            }
+        }
+
+        /// <summary>
+        /// Returns the year component of the timestamp.
+        /// </summary>
+        public int Year => _dateTime.Year;
+
+        /// <summary>
+        /// Returns the month component of the timestamp.
+        /// </summary>
+        public int Month => _dateTime.Month;
+
+        /// <summary>
+        /// Returns the day component of the timestamp.
+        /// </summary>
+        public int Day => _dateTime.Day;
+
+        /// <summary>
+        /// Returns the hour component of the timestamp.
+        /// </summary>
+        public int Hour => _dateTime.Hour;
+
+        /// <summary>
+        /// Returns the minute component of the timestamp.
+        /// </summary>
+        public int Minute => _dateTime.Minute;
+
+        /// <summary>
+        /// Returns the second component of the timestamp.
+        /// </summary>
+        public int Second => _dateTime.Second;
+
+        /// <summary>
+        /// Returns the microsecond component of the timestamp.
+        /// </summary>
+        public int Microsecond => (int)(_dateTime.Ticks -
+            new DateTime(Year, Month, Day, Hour, Minute, Second).Ticks) / 10;
+
+        /// <summary>
+        /// Readable string representation for this type.
+        /// </summary>
+        public override string ToString() => _dateTime.ToUniversalTime()
+            .ToString("yyyy-MM-dd HH:mm:ss.ffffff");
+
+        /// <summary>
+        /// Checks if the given object is same as the current object.
+        /// </summary>
+        /// <param name="obj">Other object to compare against</param>
+        /// <returns>True if the other object is equal</returns>
+        public override bool Equals(object obj) =>
+            ReferenceEquals(this, obj) ||
+            ((obj is Timestamp timestamp) && Year.Equals(timestamp.Year) &&
+                Month.Equals(timestamp.Month) && Day.Equals(timestamp.Day) &&
+                Hour.Equals(timestamp.Hour) && Minute.Equals(timestamp.Minute) &&
+                Second.Equals(timestamp.Second) && Microsecond.Equals(timestamp.Microsecond));
+
+        /// <summary>
+        /// Returns the hash code of the current object.
+        /// </summary>
+        /// <returns>The hash code of the current object</returns>
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        /// Returns DateTime object describing this type.
+        /// </summary>
+        /// <returns>DateTime object of the current object</returns>
+        public DateTime ToDateTime() => _dateTime;
+
+        /// <summary>
+        /// Returns a double object that represents the number of seconds from the epoch of
```

Comment:
Can you convey the intention that the unit is seconds but we preserve the sub-seconds as fraction?

Reply:
Sure, I have changed the summary and let me know if this makes sense.